configuration/configuration_test.go

```diff
@@ -0,0 +1,181 @@
+
+const minValidConf = `---
+configure:
+    scheduler:
+        storage_uri: /etc/ciao/configuration.yaml
+    controller:
+        compute_ca: /etc/pki/ciao/compute_ca.pem
+        compute_cert: /etc/pki/ciao/compute_key.pem
+        identity_user: controller
+        identity_password: ciao
+    launcher:
+        compute_net: 192.168.1.110
+        mgmt_net: 192.168.1.111
+    image_service:
+        url: http://glance.example.com
+    identity_service:
+        url: http://keystone.example.com
+`
+
+func TestGetPayload(t *testing.T) {
+	var conf payloads.Configure
+
+	fillDefaults(&conf)
+	conf.Configure.Scheduler.ConfigStorageURI = storageURI
+	conf.Configure.Controller.HTTPSCACert = httpsCACert
+	conf.Configure.Controller.HTTPSKey = httpsKey
+	conf.Configure.Controller.IdentityPassword = identityPassword
+	conf.Configure.Launcher.ComputeNetwork = computeNet
+	conf.Configure.Launcher.ManagementNetwork = mgmtNet
+	conf.Configure.ImageService.URL = glanceURL
+	conf.Configure.IdentityService.URL = keystoneURL
+
+	payload, _ := getPayload(&conf)
+	if payload != nil {
+		t.Fatalf("expected nil, got [%s]", string(payload))
+	}
+
+	conf.Configure.Controller.IdentityUser = identityUser
+
+	_, err := getPayload(&conf)
+	if err != nil {
+		t.Fatalf("expected nil, got %v", err)
+	}
+
+}
+
+func TestConfigurationPayload(t *testing.T) {
+	var conf payloads.Configure
+	err := ConfigurationPayload(nil, &conf)
+	if err != nil {
+		t.Fatalf("Expected nil, got %v", err)
+	}
+}
+
+func saneDefaults(conf *payloads.Configure) bool {
+	return (conf.Configure.Scheduler.ConfigStorageType == payloads.Filesystem &&
+		conf.Configure.Controller.ComputePort == 8774 &&
+		conf.Configure.ImageService.Type == payloads.Glance &&
+		conf.Configure.IdentityService.Type == payloads.Keystone &&
+		conf.Configure.Launcher.DiskLimit == true &&
+		conf.Configure.Launcher.MemoryLimit == true)
+}
+
+func TestFillDefaults(t *testing.T) {
+	var conf payloads.Configure
+	fillDefaults(&conf)
+	res := saneDefaults(&conf)
+	if res != true {
+		t.Fatalf("Expected true, got %v", res)
+	}
+}
+
+func TestValidMinConf(t *testing.T) {
+	var conf payloads.Configure
+
+	conf.Configure.Scheduler.ConfigStorageURI = storageURI
+
+	conf.Configure.Controller.HTTPSCACert = httpsCACert
+	conf.Configure.Controller.HTTPSKey = httpsKey
+	conf.Configure.Controller.IdentityUser = identityUser
+	conf.Configure.Controller.IdentityPassword = identityPassword
+
+	conf.Configure.Launcher.ComputeNetwork = computeNet
+	conf.Configure.Launcher.ManagementNetwork = mgmtNet
+
+	conf.Configure.ImageService.URL = glanceURL
+
+	valid := validMinConf(&conf)
+	if valid != false {
+		t.Fatalf("Expected false, got %v", valid)
+	}
+
+	// missing value to get minimal valid Configuration
+	conf.Configure.IdentityService.URL = keystoneURL
+
+	valid = validMinConf(&conf)
+	if valid != true {
+		t.Fatalf("Expected true, got %v", valid)
+	}
+}
+
+func TestConfigurationBlob(t *testing.T) {
```

Comment:
So this one looks quite good because it really tests many cases and does both negative and positive testing.
However, I think you should split it in several smaller tests. You can have a generic function:
```
func testConfigurationBlob(t *testing.T, uri string, expectedBlob []byte, bool positive) {
       blob, err := ConfigurationBlob(uri)
       if positive == true && err != nil {
               t.Fatalf("%s", err)
       }

       if positive == false && err == nil {
               t.Fatalf("%s", err)
       }

       if positive == true && expectedBlob != nil {
               if expectedBlob != blob {
                       t.Fatalf("Expected %v Got %v", expectedBlob, blob)
               }
       }
}
```